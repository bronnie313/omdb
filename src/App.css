* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;

  /* outline: 1px solid red; */
}

.container {
  text-align: center;
  margin-top: 4rem;
}

.container-2 {
  text-align: center;
}

.movies {
  max-width: 90vw;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  gap: 2rem;
  margin: 3rem auto;
}

.movies li {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #fff;
  padding: 1rem 2rem;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
}

.movies img {
  width: 200px;
  height: 55vh;
}

ul {
  list-style: none;
}

input[type='search'] {
  width: 30%;
  margin-right: 0.5rem;
}

a {
  text-decoration: none;
  color: inherit;
}

.icon {
  font-size: 2rem;
  position: absolute;
  top: 1rem;
  left: 1rem;
}

.details {
  margin-top: 5rem;
  padding: 0 2rem;
}

.info {
  display: flex;
  flex-direction: column;
  gap: 20px;
}

@media screen and (min-width: 768px) {
  .details {
    max-width: 90vw;
    gap: 2rem;
    margin: 2rem 15rem;
    display: flex;
    padding: 2rem;
    align-items: center;
    justify-content: space-around;
    background: #fff;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
  }

  .details-wrapper {
    width: 100%;
    display: flex;
    align-items: center;
    padding: 1rem 2rem;
    gap: 50px;
    justify-content: space-between;
  }

  .details img {
    width: 250px;
    height: 70vh;
  }

  .info {
    display: flex;
    flex-direction: column;
    gap: 20px;
  }
}
