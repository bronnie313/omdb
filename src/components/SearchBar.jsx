import React from 'react'
import { useContext, useState } from 'react'
import { SearchContext } from '../App'

const SearchBar = () => {
  const { search, setSearch } = useContext(SearchContext)

  const handleSubmission = (e) => {
    e.preventDefault()
    console.log(search)
  }

  const handleChange = (e) => {
    e.preventDefault()
    setSearch(e.target.value)
  }

  return (
    <div className="container">
      <form onSubmit={handleSubmission}>
        <h2 style={{ margin: '1rem 0' }}>OMDB MOVIES</h2>
        <input
          type="text"
          value={search}
          name="search"
          id="search"
          onChange={handleChange}
          placeholder="Search for movies"
          style={{ padding: '4px', marginRight: '1rem', width: '30%' }}
        />
        <button type="submit" style={{ padding: '4px' }}>
          search
        </button>
      </form>
    </div>
  )
}

export default SearchBar
