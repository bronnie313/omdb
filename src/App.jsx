import './App.css'
import MovieCard from './components/MovieCard.jsx'
import MovieDetails from './components/MovieDetails.jsx'
import MovieList from './components/MovieList.jsx'
import SearchBar from './components/SearchBar.jsx'

function App() {
  return (
    <>
      <MovieList />
      <MovieCard />
      <MovieDetails />
      <SearchBar />
    </>
  )
}

export default App
