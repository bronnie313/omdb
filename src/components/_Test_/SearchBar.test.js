import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import SearchBar from '../SearchBar.jsx'
import { SearchContext } from '../../App.jsx'

describe('SearchBar component', () => {
  const mockSearch = 'Avengers'
  const setSearch = jest.fn()

  test('renders SearchBar correctly', () => {
    const { container } = render(
      <SearchContext.Provider value={{ search: '', setSearch: () => {} }}>
        <SearchBar />
      </SearchContext.Provider>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('calls setSearch function with correct search value when input changes', () => {
    render(
      <SearchContext.Provider value={{ search: '', setSearch }}>
        <SearchBar />
      </SearchContext.Provider>
    )
    const input = screen.getByPlaceholderText('Search for movies')
    fireEvent.change(input, { target: { value: mockSearch } })
    expect(setSearch).toHaveBeenCalledWith(mockSearch)
  })
})
