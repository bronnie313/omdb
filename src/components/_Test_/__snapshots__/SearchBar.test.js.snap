// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders SearchBar correctly 1`] = `
<div
  class="container"
>
  <form>
    <h2
      style="margin: 1rem 0px;"
    >
      OMDB MOVIES
    </h2>
    <input
      id="search"
      name="search"
      placeholder="Search for movies"
      style="padding: 4px; margin-right: 1rem; width: 30%;"
      type="text"
      value=""
    />
    <button
      style="padding: 4px;"
      type="submit"
    >
      search
    </button>
  </form>
</div>
`;
